ofrak_core/test_ofrak/components/test_xattr_stub.py

```diff
@@ -0,0 +1,196 @@
+import os
+import xattr
+
+import pytest
+
+import ofrak.core.xattr_stub as xattr_stub
+
+EXAMPLE_DIRECTORY = os.path.join(os.path.dirname(__file__), "assets/")
+
+
+@pytest.fixture(scope="session")
+def move_to_test_directory():
+    current_directory = os.getcwd()
+    os.chdir(EXAMPLE_DIRECTORY)
+    yield
+    os.chdir(current_directory)
+
+
+@pytest.fixture
+async def xattr_stub_fixture():
+    return xattr_stub.xattr(None)
+
+
+@pytest.fixture
+async def xattr_real_fixture():
+    return xattr.xattr(os.path.join(EXAMPLE_DIRECTORY, "README.md"))
+
+
+async def test_init(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.__init__(None)
+    assert "Function __init__" in caplog.text
+    assert type(value) == type(xattr_real_fixture.__init__(None))
+
+
+async def test_repr(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.__repr__()
+    assert "Function __repr__" in caplog.text
+    assert type(value) == type(xattr_real_fixture.__repr__())
+
+
+async def test_call(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture._call(None, None)
+    assert "Function _call" in caplog.text
+    # Not testing return type of _call because it is dependent on the function being called. Could
+    # be either str or None. _call should only be called indirectly through methods like get()
+    # or set(), which are being type tested.
+
+
+async def test_get(xattr_stub_fixture, xattr_real_fixture, caplog):
+    xattr_real_fixture.set("user.foo", b"bar")
+    value = xattr_stub_fixture.get(None)
+    assert "Function get" in caplog.text
+    assert type(value) == type(xattr_real_fixture.get("user.foo"))
+
+
+async def test_set(xattr_stub_fixture, xattr_real_fixture, caplog):
+    value = xattr_stub_fixture.set(None, None)
```

Comment:
It is confusing here (and later on) that the stub fixture is getting passed different arguments than the real fixture, since it is unclear what is being tested.

My understanding for this and other tests in this file is, that the test is the following:

_Run the method on stub and real xattr with same arguments, and validate that the return type matches_

Making the arguments the same will make this intention clearer.